pkg/controllermanager/controller/shoot/shoot_maintenance_control.go

```diff
@@ -224,3 +219,81 @@ func hasMaintainNowAnnotation(shoot *gardenv1beta1.Shoot) bool {
 	operation, ok := shoot.Annotations[common.ShootOperation]
 	return ok && operation == common.ShootOperationMaintain
 }
+
+// MaintainMachineImage determines if a shoots machine image has to be maintained and in case returns the target image
+func MaintainMachineImage(shoot *gardenv1beta1.Shoot, cloudProfile *gardenv1beta1.CloudProfile, shootCurrentImage *gardenv1beta1.MachineImage) (bool, *gardenv1beta1.MachineImage, error) {
+	machineImagesFound, machineImages, err := helper.DetermineMachineImages(*cloudProfile, shootCurrentImage.Name)
+	if err != nil {
+		return false, nil, fmt.Errorf("failure while determining the machine images in the CloudProfile: %s", err.Error())
+	}
+	if !machineImagesFound || len(machineImages) == 0 {
+		return false, nil, fmt.Errorf("failure while determining the machine images in the CloudProfile: no machineImage with name '%s' (specified in shoot) could be found in the cloud cloudProfile '%s'", shootCurrentImage.Name, cloudProfile.Name)
+	}
+
+	// checks if forceUpdate Time & applicable (has the specified version)
+	forceUpdateRequired, forceUpdateTime := ForceUpdateRequired(shootCurrentImage, machineImages)
+
+	// safely dereference due to defaulting in api server
+	autoUpdateOfMachineImageConfigured := *shoot.Spec.Maintenance.AutoUpdate.MachineImageVersion
+	if forceUpdateRequired || autoUpdateOfMachineImageConfigured {
+		versionUpdatePossible, updateMachineImage, err := VersionUpdatePossible(shootCurrentImage, machineImages)
+		if err != nil {
+			return false, nil, nil
+		}
+		if forceUpdateRequired && !versionUpdatePossible {
```

Comment:
I'd prefer to ignore this case.

Reply:
This is a configuration mistake of the CloudProfile's machine images. However we cannot check it during update of the CloudProfile. 
I thought by throwing an error we could indicate that the auto update was not possible due to a configuration problem. 
Would you let the admin know that there is a config problem in the CloudProfile? And if yes, how?

-> possibly an event on the CloudProfile?

Reply:
Sorry, I was confused by the term `ForceUpdateTime`.

> However we cannot check it during update of the CloudProfile.

Why is that? Can't we assume that the latest version of an image in the `CloudProfile` must not have a `ForceUpdateTime` aka `ExpirationDate`?

Reply:
Yep, we could do it like this or (like in the current implementation) give the admin the freedom to add another machine image at a later point in time.
Which variant would you prefer? I am fine with either of them.

Reply:
> Yep, we could do it like this or (like in the current implementation) give the admin the freedom to add another machine image at a later point in time.

Can you elaborate on that?
My assumption was that the admin is able to do so anyway.

Let's say we first have:
```
machineImages:
- name: coreos
  version: 1.0
```

And after a while the admin updates the `CloudProfile`:

```
machineImages:
- name: coreos
  version: 2.0
- name: coreos
  version: 1.0
  expirationDate: 2019-08-01T00:00:00Z
```

Reply:
We could either allow that the latest version has an expiration date or not. 
But thinking about it again I agree that it would prevent configuration issues in the cloud profile if we enforce that the latest machine image does not have an expiration date.
I added a check in the storage and adjusted the tests.